client/packages/state/reducer.js

```diff
@@ -83,7 +113,27 @@ reducers[ TOGGLE_OUTER_DIMENSIONS ] = ( state ) => {
 	} );
 };
 
+reducers[ REMOVE_PACKAGE ] = ( state, action ) => {
+	const packages = ( state.packages || [] ).slice();
+	packages.splice( action.index, 1 );
+	return Object.assign( {}, state, {
+		packages,
+		pristine: false,
+	} );
+};
+
+reducers[ SET_IS_SAVING ] = ( state, action ) => {
+	return Object.assign( {}, state, {
+		isSaving: action.isSaving,
+		pristine: ! action.isSaving, //set pristine after the form has been saved
+	} );
+};
+
 const packages = ( state = {}, action ) => {
+	if ( 'undefined' === typeof state ) {
```

Comment:
This is never going to be true, because you put `{}` as default value for the `state` argument. I think what you are trying to do is this:
`const packages = ( state = initialState, action ) => {`